src/test/java/com/conveyal/datatools/manager/jobs/ArbitraryTransformJobTest.java

```diff
@@ -0,0 +1,192 @@
+import com.conveyal.datatools.manager.models.transform.ReplaceFileTransformation;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import static com.conveyal.datatools.TestUtils.assertThatSqlCountQueryYieldsExpectedCount;
+import static com.conveyal.datatools.TestUtils.createFeedVersionFromGtfsZip;
+import static com.conveyal.datatools.manager.models.FeedSource.FeedRetrievalMethod.MANUALLY_UPLOADED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+public class ArbitraryTransformJobTest extends UnitTest {
+    private static final Logger LOG = LoggerFactory.getLogger(ArbitraryTransformJob.class);
+    private static Auth0UserProfile user = Auth0UserProfile.createTestAdminUser();
+    private static Project project;
+    private static FeedSource bart;
+    private FeedVersion bartVersion;
+    private FeedVersion newBartVersion;
+
+    /**
+     * Initialize Data Tools and set up a simple feed source and project.
+     */
+    @BeforeClass
+    public static void setUp() throws IOException {
+        // start server if it isn't already running
+        DatatoolsTest.setUp();
+
+        // Create a project, feed sources, and feed versions to merge.
+        project = new Project();
+        project.name = String.format("Test %s", new Date().toString());
+        Persistence.projects.create(project);
+
+        // Bart
+        bart = new FeedSource("BART", project.id, MANUALLY_UPLOADED);
+        Persistence.feedSources.create(bart);
+    }
+
+    /**
+     * Clean up test database after tests finish.
+     */
+    @AfterClass
+    public static void tearDown() {
+        // Project delete cascades to feed sources.
+        project.delete();
+    }
+
+    /**
+     * Run set up before each test. This just resets the feed source transformation properties.
+     */
+    @Before
+    public void setUpTest() {
+        bart = Persistence.feedSources.getById(bart.id);
+        bart.transformRules = new ArrayList<>();
+        Persistence.feedSources.replace(bart.id, bart);
+    }
+
+    /**
+     * Run tear down after each test. This just deletes the feed versions that were used in the test.
+     */
+    @After
+    public void tearDownTest() {
+        // Clean up
+        if (bartVersion != null) bartVersion.delete();
+        if (newBartVersion != null) newBartVersion.delete();
+    }
+
+    @Test
+    public void canReplaceGtfsPlusFile() throws IOException {
```

Comment:
Would be useful to have some JavaDoc explaining the process and goal of this test case.

Reply:
This test should be in its own file.

Reply:
I think there should be a file for all the transformations tested.